superset/migrations/shared/utils.py

```diff
@@ -185,15 +193,211 @@ def has_table(table_name: str) -> bool:
+            logger.info(
+                f"Column {LRED}{col.name}{RESET} already present on table {LRED}{table_name}{RESET} Skipping..."
+            )
+        else:
+            cols_to_add.append(col)
+
+    with op.batch_alter_table(table_name) as batch_op:
+        for col in cols_to_add:
+            logger.info(
+                f"Adding column {GREEN}{col.name}{RESET} to table {GREEN}{table_name}{RESET}"
+            )
+            batch_op.add_column(col)
+
+
+def drop_columns(table_name: str, *columns: str) -> None:
+    """
+    Drops specified columns from an existing database table.
+
+    For each column, it first checks if the column exists in the table. If a column does not exist, it logs an informational
+    message and skips the dropping of that column. Otherwise, it proceeds to remove the column from the table.
+
+    The operation is performed within a batch operation context which allows a more performant approach
+
+    :param table_name: The name of the table from which the columns will be removed.
+    :param columns: A list of column names to be dropped.
+    """
+
+    cols_to_drop = []
+    for col in columns:
+        if not table_has_column(table_name=table_name, column_name=col):
+            logger.info(
+                f"Column {LRED}{col}{RESET} is not present on table {LRED}{table_name}{RESET} Skipping..."
+            )
+        else:
+            cols_to_drop.append(col)
+
+    with op.batch_alter_table(table_name) as batch_op:
+        for col in cols_to_drop:
+            logger.info(
+                f"Dropping column {GREEN}{col}{RESET} from table {GREEN}{table_name}{RESET}"
+            )
+            batch_op.drop_column(col)
+
+
+def create_index(table_name: str, index_name: str, *columns: str) -> None:
+    """
+    Creates an index on specified columns of an existing database table.
+
+    This function checks if an index with the given name already exists on the specified table.
+    If so, it logs an informational message and skips the index creation process.
+    Otherwise, it proceeds to create a new index with the specified name on the given columns of the table.
+
+    The operation is performed within a batch operation context which allows a more performant approach
+
+    :param table_name: The name of the table on which the index will be created.
+    :param index_name: The name of the index to be created.
+    :param columns: A list column names where the index will be created
+    """
+
+    if table_has_index(table=table_name, index=index_name):
+        logger.info(
+            f"Table {LRED}{table_name}{RESET} already has index {LRED}{index_name}{RESET} Skipping..."
+        )
+        return
+
+    logger.info(
+        f"Creating index {GREEN}{index_name}{RESET} on table {GREEN}{table_name}{RESET}"
+    )
+
+    with op.batch_alter_table(table_name) as batch_op:
+        batch_op.create_index(index_name=index_name, columns=columns)
```

Comment:
Do you need to use a batch operation here given that it's a single operation? Same for `drop_index`.

Reply:
Updated on [commit](https://github.com/apache/superset/pull/31303/commits/5560992f36d18258dc3e394118ca50e8eed94464)